packages/markers/src/browser/problem/problem-widget.tsx

```diff
@@ -81,6 +81,22 @@ export class ProblemWidget extends TreeWidget {
         return 'caption';
     }
 
+    protected renderTailDecorations(node: TreeNode, props: NodeProps) {
+        return <div className='row-button-container'>
+            {this.renderRemoveButton(node)}
+        </div>;
+    }
+
+    protected renderRemoveButton(node: TreeNode): React.ReactNode {
+        return <span className='remove-node' onClick={e => this.remove(node, e)}></span>;
```

Comment:
`onClick={this.remove}` - there is no point to store `remove` function as a property, if you create a new one on each rendering